### ISSUE TITLE
Dual monitor setup not working?

### ISSUE THREAD

**Issue by legostax** [awesomeWM/awesome#2002]
Output of `awesome --version`:
```
awesome v4.2 (Human after all)
 • Compiled against Lua 5.1.5 (running with Lua 5.1)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.4
 • LGI version: 0.9.0
```

**Background:**
I'm fairly certain that this is an Awesome-specific issue because I have been able to get both monitors to work with Xfce.  The monitors are connected via DisplayPort to a GTX 1070 running the proprietary NVIDIA drivers.  The primary screen works just fine.  Awesome displays on it as expected.  I'd like to note that this problem persists with the default `rc.lua`.

**Problem:**
Awesome does not seem to see the secondary screen at all.  I am able to move my mouse over it, but the cursor changes to this: ![x-shaped cursor](http://i.imgur.com/skaM0SP.gif)  I cannot move clients over to the secondary screen.
When I run
`naughty.notify({text = "2000:1: "..tostring(awful.screen.getbycoord(2000,1))})`
the following is printed out:
`2000:1: 1`
Shouldn't that say `2000:1: 2` if Awesome saw the secondary screen?  Perhaps I'm clueless and am doing this wrong, which is probably likely.

**Comment by Elv13** [comment]
Awesome does not manage physical screens. While XFCE/KDE/Gnome have some utilities to do it automatically, Awesome doesn't. It is out of scope for the project. You need to use `xrandr` to enable the screen. There is some nice little graphical frontend like `arandr` that will help you craft the correct command. You can then place it in a script. Once the screen is enabled, Awesome will detect it correctly.

**Comment by nickdiego** [comment]
How did you enable the secondary display?
xrandr? could you provide the output for `xrandr -q`?

**Comment by legostax** [comment]
I appreciate the fast responses!
`xrandr -q`
```
Screen 0: minimum 8 x 8, current 1920 x 1080, maximum 32767 x 32767
DVI-D-0 disconnected primary (normal left inverted right x axis y axis)
HDMI-0 disconnected (normal left inverted right x axis y axis)
DP-0 disconnected (normal left inverted right x axis y axis)
DP-1 disconnected (normal left inverted right x axis y axis)
DP-2 connected 1920x1080+0+0 (normal left inverted right x axis y axis) 531mm x 299mm
   1920x1080     60.00 + 144.00*  119.98    99.93    84.90    59.94    50.00    60.00    50.04  
   1680x1050     59.95  
   1440x900      59.89  
   1280x1024     75.02    60.02  
   1280x960      60.00  
   1280x800      59.81  
   1280x720      60.00    59.94    50.00  
   1152x864      75.00  
   1024x768      75.03    70.07    60.00  
   800x600       75.00    72.19    60.32    56.25  
   720x576       50.00  
   720x480       59.94  
   640x480       75.00    59.94    59.93  
DP-3 disconnected (normal left inverted right x axis y axis)
DP-5 disconnected (normal left inverted right x axis y axis)
```

**Comment by nickdiego** [comment]
As @Elv13 said, awesome does not manage the X screens itself.. and something seems to be wrong at the Xorg level, if you have a secondary monitor physically connected, it should appear as connected in `xrandr -q` output, i guess.. You can try something like `xrandr --auto` to test if it works, or you can try to use some other xrandr frontend such as `arandr`, as mentioned before.

Anyway, it does not seem to be an awesome issue.

**Comment by legostax** [comment]
Yeah, that's what I'm finding...  Good to know.  As a noob, I appreciate the help!  This was probably a question for superuser.com anyway.  I'll continue to dig in config files and settings.

**Comment by legostax** [comment]
For everyone who comes here later, trying to figure out problems with their new dual monitor setup:

**Background:**
I started Xfce played around with the `nvidia-settings` app to attempt to get both monitors to work, figuring that would be all I needed to do.  This, in fact, actually caused my problem.  When I clicked **Save to X Configuration File**, I was allowing the proprietary drivers to mess up my configuration.  Only after 2 hours of research, playing around, and reboots did I find this out.  Do **NOT** use `nvidia-settings` to set things up.  Use `xrandr`.  At the time of filing this issue, I had no idea what `xrandr` was do my 'noobiness'.

**Fix:**
1. If you're on Ubuntu, open your `/etc/X11/xorg.conf` and remove all lines and sections that deal with setting up your second monitor.  Save and reboot or restart X.
2. Run `xrandr -q` to query all of the available inputs.  If you were only seeing one connected monitor, and now you see two, continue along.  If you still see only one connected monitor, this fix is obviously not for you.
3. I wanted to have my primary monitor (identified in the `xrandr -q` by `DP-2`) on the left, my secondary monitor (identified by `DP-4`) on the right.  I wanted both monitors to be set to 1080p at 144hz.  Thus, my xrandr command to run was:
```
xrandr --output DP-2 --mode 1920x1080 --rate 144.00 --primary --output DP-4 --right-of DP-2 --mode 1920x1080 --rate 144.00
```
For you, the reader, you would need to run the following, putting in your own settings:
```
xrandr --output [the identifier for your first monitor] --mode [the max resolution of your first monitor] -- rate [refresh rate of your first monitor] --primary --output [second monitor] --right-of [first monitor] --mode [max resolution of second monitor] -- rate [refresh rate of second monitor]
```
Hopefully it is obvious how you need to change the command to match your setup.  The other options that correspond to `--right-of` can be found by running `xrandr --help`.
